### DISCUSSION TITLE
Generic F-droid warning about a network related "Anti-Feature"

### DISCUSSION THREAD

**Discussion by InsertUser** [streetcomplete/StreetComplete#5756] [General]
The F-Droid [page](https://f-droid.org/packages/de.westnordost.streetcomplete/) for StreetComplete currently complains about a very vague "Anti-Feature":

> This app promotes or depends entirely on a non-changeable or non-free network service

Does anyone know what this is?

It seems like really bad behaviour on their part to make such a broad "warning" without giving any way to see what the problem supposedly is (that I can see at least).

**Comment by matkoniecz** [comment]
It complains that OpenStreetMap editor can be used only for editing OpenStreetMap. 

This is in my opinion really pointless and gave feedback to them about it at some point. 

 We do not control fdroid

**Reply by InsertUser** [reply]
According to [this][1] lemmy comment they're able to see more detail:

> Tethered to commercial Jawg.io for map tiles, FOSS OpenstreetMap.org and Westnordost.de to get quests, upload data, etc

I don't know why they are able to get an explanation that doesn't show for me in Firefox. 

I thought westnordost.de was only being used for photos due to the lack of suitable alternatives and as that is optional anyway throwing a warning for that seems really crummy. 

Complaining about a hardcoded tile service is just bizarre. Does every single travel app on F-Droid carry this warning?

[1]: https://lemmy.ml/comment/12424857

**Reply by matkoniecz** [reply]
> westnordost.de was only being used for photos due to the lack of suitable alternatives 

It is also used for statistics and version check (some old version are blocked due to be damaging data)

Also, it is open source.

**Reply by matkoniecz** [reply]
> Complaining about a hardcoded tile service is just bizarre. Does every single travel app on F-Droid carry this warning?

Actually, this one would make some sense to be configurable, at least for some people and Jawg is actually proprietary from what I remember. 

But for example complaining that OpenStreetMap editor is for editing OpenStreetMap seems quite unhelpful.

**Comment by mnalis** [comment]
> Does anyone know what this is?

Yes, that `NonFreeNet` warning is too broad, see below about fixes in the works.... 
It's [current definition](https://gitlab.com/fdroid/fdroiddata/-/blob/master/config/antiFeatures.yml) is _"This app promotes or depends entirely on a non-changeable **or** non-free network service"_

Note that all antifeatures in F-droid are intended as useful pointers for **end user** to make an **informed choice that suits them the most**; so it is about easy transparency. There are not _"warnings"_. They are not about how good/bad/moral software is (that is left to user discretion; otherwise there would be hundred different opinions what F-droid should mark as good/bad[^1]

**All software**  that is **distributed on F-droid** is always **fully FOSS**, so if whether the client you're installing is free software / open source  is only thing you care about, you can ignore all antifeature flags.

There is F-droid work to split it into [two separate Antifeatures](https://gitlab.com/fdroid/fdroid-website/-/merge_requests/946), one of which (original `NonFreeNet`) would be about "non-free" and another `TetheredNet` about  "non-changeable" parts. That way: 

- FOSS client to access for example `Facebook.com` or `x.com` would be `NonFreeNet`.
  The information conveyed to the user being that while the client they're using is open source, the [SPOF](https://en.wikipedia.org/wiki/Single_point_of_failure) server is not, and if owners of server ever decides to do so, they might lose all their data/access/social networking/invested effort/etc. without recourse. Thus, they might want to choose to use some other service (like [Fediverse](https://en.wikipedia.org/wiki/Fediverse)) which allows them more exit options (See [Vendor lock-in](https://en.wikipedia.org/wiki/Vendor_lock-in))

- OSM editor that only has hardcoded dependency on `openstreetmap.org` would get milder `TetheredNet` antifeature (since user cannot change it to use some other instance using same API, like [fosm.org](https://wiki.openstreetmap.org/wiki/Fosm.org) CC-BY-SA fork or [OHM](https://wiki.openstreetmap.org/wiki/OpenHistoricalMap) or [testing APIs](https://apis.dev.openstreetmap.org/) or even their own instance).
  The information conveyed to the user being that if the service ever closes/stops being acceptable to the user (e.g. OSM bought out by Facebook, or license changed, or political decisions contrary to the user viewpoint etc), the user would have to become programmer to use some of the other compatible instances, instead of just changing a setting in their app. And that could be a high bar (depending on list of dependencies, [bitrot](https://en.wikipedia.org/wiki/Software_rot), etc.)

---

Current [antifeature details text for StreetComplete](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/de.westnordost.streetcomplete.yml#L1-L4) is _"Tethered to commercial Jawg.io for map tiles, FOSS OpenStreetMap.org and Westnordost.de to get quests, upload data, etc."_

So idea is to be fully transparent to the user with short but succinct details. If you are fine with those above, you're good, enjoy the app! If you dislike some of those (e.g. perhaps you are totally anti-commercial and want to use some other vector tile provider than Jawg, or your country is being blocked by Germany servers and you are huge user of picture notes, or whatever), those should provide you with enough data to make informed choice. 

IOW, F-Droid requires **users** to think for themselves what is acceptable to **them**; it does not serve them one-size-fits-all morality on a plate ready for consumption. (One might like that approach or not, of course, depending where they land on choice-vs-prechewed preference scale. I personally tend toward "choice" end)

> I don't know why they are able to get an explanation that doesn't show for me in Firefox.

Yes, That is quite unfortunate: extra explanations for antifeatures (part of `index-v2`) are **currently implemented in mobile client only**, but not yet in a web. See [f-droid website #770](https://gitlab.com/fdroid/fdroid-website/-/issues/770) for that. Help is more than welcome!


[^1]:  e.g. _"are ads always bad"_, _"how about if ads income is only thing allowing that FOSS devel to work on the app"_, _"is anything connecting to facebook or X always evil"_, _"is vendor lock-in always bad"_, _"is displaying nudity / violence ever ok"_,  _"are open source apps that contain images which may not be changed / resold OK"_, _"should apps which have had no open-source updates for years good"_, _"what about if app was opensource but has turned proprietary"_ etc) .

**Reply by InsertUser** [reply]
> There are not "warnings". They are not about how good/bad/moral software is (that is left to user discretion; otherwise there would be hundred different opinions what F-droid should mark as good/bad[1](https://github.com/streetcomplete/StreetComplete/discussions/5756#user-content-fn-1-a51aaadd1c9eb3c2777314b2aa492f89)

They clearly [_are_](https://www.collinsdictionary.com/dictionary/english/warning) warnings, however they happen to have been rebranded. 

> Yes, That is quite unfortunate: extra explanations for antifeatures (part of index-v2) are currently implemented in mobile client only, but not yet in a web. See [f-droid website #770](https://gitlab.com/fdroid/fdroid-website/-/issues/770) for that. Help is more than welcome!

Now _that's_ what I'd call an Anti-Feature. They'll only tell you what's wrong with something if you let them run native code? That's a massive red flag.

**Reply by mnalis** [reply]
> Now that's what I'd call an Anti-Feature. 

Well, that is a **missing** feature, not an anti-feature, though...

> They'll only tell you what's wrong with something if you let them run native code? 

_Or_ if you invest **1 extra click** on that **Build Metadata** link (which I also provided above) if you prefer to use the web, and want details.

Yes, I **absolutely agree with you that it would be much nicer if that information was available without any extra clicks and nicely formatted**. But that feature is currently _missing_, because prerequisite for supporting it (implementing `index-v2` parsing IIUC) is not trivial (were it trivial, I'd learn enough ruby to make MR adding that). 

_(In fact, it was me that opened that issue there requesting web support for antifeature details, so I'm on your side here, no need for an offensive :smiley_cat:)_

> That's a massive red flag.

You are entitled to your opinion, of course. The correct place to complain about that and ask to be fixed would be on their GitLab, especially on that [issue 770](https://gitlab.com/fdroid/fdroid-website/-/issues/770) (which I also linked previously), instead of StreetComplete discussions (where probably nobody relevant will ever even see it) :man_shrugging: 

Anyway, I'm not F-droid maintainer nor their spokesperson, just a perhaps slightly-above-average-informed user and small-patch contributor (just like I'm on StreetComplete), so I tried (in good will) to explain the things (as I see them) that you asked about.

I'm not making their policies nor have a say in them any more than you or any other user. So, if you want it fixed, you're welcome to help. If you can code in jekyll/ruby, hey, I'll owe you a beer if you fix that! If you can't code that (like I can't), you can still at least add your supporting comment explaining why you find it important in that ticket, add a :+1:,  and perhaps even ask other interested users in your social circle that care about the issue to chime in there. I've noticed that issues that people show _more interest in_ get resolved more quickly than those that seem low priority and without interested users...

**Reply by InsertUser** [reply]
> Or if you invest 1 extra click on that Build Metadata link (which I also provided above) if you prefer to use the web, and want details.

If this file were linked from the same section I'd have clicked it. But what I actually clicked when looking for more info was the build log and found it not readable. The build metadata being among the links to the original source and donation pages rather than the stuff F-Droid has processed is very anti intuitive. 

When I have the time to make a new account on yet another site I might reply to the original bug asking them to do the bare minimum for usability (put an extra link somewhere sensible) .

**Reply by mnalis** [reply]
> When I have the time to make a new account on yet another site

FYI @InsertUser, if it helps, you can also login on GitLab with your GitHub account (Oauth SSO), so it is quick and easy that way.

**Reply by licaon-kter** [reply]
thanks @mnalis for the explainer :)

@InsertUser We recommend the F-Droid Client to get timely updates and info on new apps. The website is updated later and only as a convenience. Feel free to lend a hand to fix any open issues regarding the website or elsewhere. :smile:
